src/monitoring/mod.rs

```diff
@@ -117,6 +131,101 @@ pub fn increment_errors_emitted_counter() {
     prometheus::ERRORS_EMITTED_COUNTER.inc();
 }
 
+fn txid_tracking_db(chainstate_root_path: &str) -> Result<DBConn, DatabaseError> {
+    let mut path = PathBuf::from(chainstate_root_path);
+
+    path.push("tx_tracking.db");
+    let db_path = path.to_str().ok_or_else(|| DatabaseError::ParseError)?;
+
+    let mut create_flag = false;
+    let open_flags = if fs::metadata(&db_path).is_err() {
+        // need to create
+        create_flag = true;
+        OpenFlags::SQLITE_OPEN_READ_WRITE | OpenFlags::SQLITE_OPEN_CREATE
+    } else {
+        // can just open
+        OpenFlags::SQLITE_OPEN_READ_WRITE
+    };
+
+    let conn = DBConn::open_with_flags(&db_path, open_flags)?;
+
+    conn.busy_handler(Some(tx_busy_handler))?;
+
+    if create_flag {
+        conn.execute(
+            "CREATE TABLE processed_txids (txid TEXT NOT NULL PRIMARY KEY)",
+            rusqlite::NO_PARAMS,
+        )?;
+    }
+
+    Ok(conn)
+}
+
+fn txid_tracking_db_contains(conn: &DBConn, txid: &Txid) -> Result<bool, DatabaseError> {
+    let contains = conn
+        .query_row(
+            "SELECT 1 FROM processed_txids WHERE txid = ?",
+            &[txid],
+            |_row| Ok(true),
+        )
+        .optional()?
+        .is_some();
+    Ok(contains)
+}
+
+#[allow(unused_variables)]
+pub fn mempool_accepted(txid: &Txid, chainstate_root_path: &str) -> Result<(), DatabaseError> {
+    #[cfg(feature = "monitoring_prom")]
+    {
+        let tracking_db = txid_tracking_db(chainstate_root_path)?;
+
+        if txid_tracking_db_contains(&tracking_db, txid)? {
+            // processed by a previous block, do not track again
```

Comment:
Couldn't this also be taken in the event a transaction winds up in two forks?

Reply:
Yeah -- I think in that case as well, we wouldn't want the prom metrics to update. In general, our prom metrics _aren't_ fork aware. And this check just ensures that at least we aren't counting the same tx multiple times.